research/gaia/gremlin/gremlin_core/src/structure/graph.rs

```
@@ -33,8 +34,21 @@ impl<E: Element + Send + Sync> QueryParams<E> {
         self.filter = Some(Arc::new(filter))
     }
 
-    pub fn set_props(&mut self, props: Vec<PropKey>) {
-        self.props = Some(props)
+    // props specify the properties we query for, e.g.,
+    // Some(vec![prop1, prop2]) indicates we need prop1 and prop2,
+    // Some(vec![]) indicates we need all properties
+    // and None indicates we do not need any property,
+    pub fn set_props_with(&mut self, fetch_properties: Option<pb::PropKeys>) {
+        if let Some(fetch_props) = fetch_properties {
+            let mut prop_keys = vec![];
+            for prop_key in fetch_props.prop_keys {
+                prop_keys.push(PropKey::from_pb(prop_key).unwrap());
```

`unwrap()` should not be here. Can we throw the error from this function?